src/core/infer_request.cc

```diff
@@ -487,6 +487,8 @@ InferenceRequest::Normalize()
 {
   const ModelConfig& model_config = backend_raw_->Config();
 
+  is_decoupled_ = model_config.model_transaction_type().decoupled();
```

Comment:
Decoupled is a property of the model. Both request and response have a handle to the backend which in turn has the config object from which they can get the decoupled flag. If we want to have an "IsDecoupled()" utility method we should put that in the InferenceBackend class (like we do for DefaultPriorityLevel).